dsc cv-fa-team 이상헌 포스팅입니다. (#66)
DSC-University-of-Seoul/dsc-university-of-seoul.github.io: _posts/2020-12-24-Image-Translation-paper-review(vol1).md

- 셀 애니메이션(Cell Animation) : Cell 이라는 투명한 종이를 뜻하는 Celluloid 를 의미합니다. 기본적으로 하나의 배경셀과 여러 장의 전경 셀들을 여러겹의 층으로 구성하여 하나의 프레임을 형성하는 방식입니다.
- <https://animationbasketcase.wordpress.com/2015/05/27/celluloid-animation/>.
- 입력 이미지로부터 segmentation map을 추출하여 Global한 구조 정보와 sparse한 color block을 효과적으로 포착합니다.

### 2-3. texture representation

- 이 특징은 미술가들이 먼저 윤곽과 디테일한 선 스케치를 그린 뒤 색을 입히는 방법에 의해 착안했습니다.
- 3개의 채널을 가진 입력이미지를 1개의 채널로 변환하는 과정입니다.
- 색상과 밝기가 제거되고 상대적인 픽셀 강도가 보존됩니다.

    ![Capture](../assets/images/post-image-translation/paper1.png)

## 3. Proposed Approach

- 아래 그림을 보면, image 특징을 Decompose하는 3가지 방법이 어떻게 진행되는지 알 수 있습니다.

    ![Capture](../assets/images/post-image-translation/paper2.png)

- 이미지가 Surface Representation, Structure Representation, Texture Representation으로 분해되고 해당 Representation을 추출하기 위해 세 개의 독립 모듈을 도입했습니다.
- GAN framework는 Generator G와 Discriminator인 D_s, D_t로 구성됩니다.
- D_s는 Surface Representation을 기준으로, 생성된 image와 만화를 구별하고 D_t는 Texture Representation 으로 생성된 image와 만화를 구별합니다.

    ![Capture](../assets/images/post-image-translation/paper3.png)

### 3-1. Learning From the Surface Representation

- 화가들이 붓으로 초안을 그리는 스타일을 모방하여 만화와 유사한, 매끄러운 표면을 가진 화풍을 생성하고자 합니다.

>     ![Capture](../assets/images/post-image-translation/paper2-1.png)

Reviewer: image file 명을 잘못 입력하셨습니다.
```suggestion
    ![Capture](../assets/images/post-image-translation/paper2_1.png)
```

- F_dgf는 이미지 I를 입력으로 받아서 texture와 디테일한 부분을 제거한 surface를 추출합니다.
- D_s는 모델이 생성해낸 이미지와 기준이되는 만화 이미지가 비슷한 surface를 갖는지 판단하고 generator G가 surface representation를 잘 추출할 수 있도록 가이드를 제공합니다.

    ![Capture](../assets/images/post-image-translation/paper4.png)

- I_c는 reference cartoon images를, I_p는 input photo를 의미합니다.

### 3-2. Learning From the Structure representation

- felzenszwalb algorithm를 사용해서 이미지를 여러 개의 지역으로 segmentation 합니다.
- superpixel algorithms을 사용하여 픽셀들의 색 등 low level 정보를 바탕으로 비슷한 픽셀끼리 묶어서 '커다란 픽셀'을 만듭니다.

>     ![Capture](../assets/images/post-image-translation/paper2-2.png)

Reviewer: image file 명을 잘못 입력하셨습니다.
```suggestion
    ![Capture](../assets/images/post-image-translation/paper2_2.png)
```

***Ex 1: felzenszwalb algorithm***

![Capture](../assets/images/post-image-translation/paper5.png)

***Ex 2: superpixel algorithms***

![Capture](../assets/images/post-image-translation/paper6.png)

***Ex 3: Selective Search***

- <https://www.geeksforgeeks.org/selective-search-for-object-detection-r-cnn/>.

- 이후 Adaptive coloring algorithm을 사용하여 이미지에 대한 Haze를 줄입니다.
- Region의 컬러값을 평균으로 취하지는 않고, adaptive하게 결정합니다. 평균으로 할 경우 darken images, hazing effect가 발생했기 떄문입니다.
- Haze란 '대상 물체와 관찰자 사이에 존재하는 물질들에 의해 빛이 진행을 방해받아 대상이 뿌옇게 보이는 현상'을 의미합니다.  아래 그림에서 (a)가 Haze가 있는 사진, (b)는 Haze를 제거한 사진입니다.

#### Adaptive coloring algorithm

![Capture](../assets/images/post-image-translation/paper7.png)

![Capture](../assets/images/post-image-translation/paper8.png)

#### Haze 유무

> - 입력 사진은 Generator에도 들어가고 VGG 네트워크에도 들어갑니다. VGG 네트워크에서는 High level feature를 추출해냅니다. Structure 특징을 뽑아내기 위한 loss는 아래와 같습니다.
> ![Capture](../assets/images/post-image-translation/paper9.png)

Reviewer: 한줄 띄어야 합니다.
```suggestion
- 입력 사진은 Generator에도 들어가고 VGG 네트워크에도 들어갑니다. VGG 네트워크에서는 High level feature를 추출해냅니다. Structure 특징을 뽑아내기 위한 loss는 아래와 같습니다.

![Capture](../assets/images/post-image-translation/paper9.png)
```

![Capture](../assets/images/post-image-translation/paper10.png)

### 3-3) Learning From the Textural Representation

- 만화영상과 실제 사진을 구분하기 위해서는 High frequency 특징과 밝기, 색 정보를 이용해야한다. 특히 High frequency 특징 정보는 남기고 밝기와 색 정보가 주는 영향을 최소화 해야합니다.

> - 즉 real world photo와 cartoon은 color와 luminance 정보로 너무 쉽게 판별이 가능하므로, 합쳐서 흑백 이미지로 만듬 + random하게 RGB를 섞는 과정을 진행합니다.
>      ![Capture](../assets/images/post-image-translation/paper2-3.png)
> - F_rcs라는 random color shift algorithm을 제안했는데, 식은 아래와 같습니다.

Reviewer: * image file 명을 잘못 입력하셨습니다.
* 한칸씩 띄어주어야합니다.
```suggestion
- 즉 real world photo와 cartoon은 color와 luminance 정보로 너무 쉽게 판별이 가능하므로, 합쳐서 흑백 이미지로 만듬 + random하게 RGB를 섞는 과정을 진행합니다.

     ![Capture](../assets/images/post-image-translation/paper2_3.png)
     
- F_rcs라는 random color shift algorithm을 제안했는데, 식은 아래와 같습니다.
```

---
layout: post
title: "Image Translation paper review"
authors: [lsh110600]
tags: ["computer vision","deep learning"]
image: assets/images/post-image-translation/title.jpg
description: "Image Translation paper review"
featured: true
---

> Image Translation 논문 리뷰
> 
> 2020 DSC-Vision team에서 Face2Cartoon 서비스를 만들 때 참고했던 논문입니다.  
> 
> Wang, Xinrui, and Jinze Yu. "Learning to Cartoonize Using White-Box Cartoon Representations." Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition. 2020.
> <https://github.com/SystemErrorWang/White-box-Cartoonization/>.
> 에 대해서 리뷰합니다.

Reviewer: 논문을 각주로 넣는게 좋을 것 같아요.
```suggestion
2020 DSC-Vision team에서 Face2Cartoon 서비스를 만들 때 참고했던 [Image Translation 관련 논문](https://github.com/SystemErrorWang/White-box-Cartoonization/)입니다.  

> Wang, Xinrui, and Jinze Yu. "Learning to Cartoonize Using White-Box Cartoon Representations." Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition. 2020.
```
